src/main/java/de/hysky/skyblocker/skyblock/slayers/SlayerEntitiesGlow.java

```diff
@@ -0,0 +1,116 @@
+    }
+
+    public static Box getSlayerMobBoundingBox(LivingEntity entity) {
+        return switch (SlayerUtils.getSlayerType()) {
+            case SlayerUtils.REVENANT ->
+                    new Box(entity.getX() - 0.4, entity.getY() - 0.1, entity.getZ() - 0.4, entity.getX() + 0.4, entity.getY() - 2.2, entity.getZ() + 0.4);
+            case SlayerUtils.TARA ->
+                    new Box(entity.getX() - 0.9, entity.getY() - 0.2, entity.getZ() - 0.9, entity.getX() + 0.9, entity.getY() - 1.2, entity.getZ() + 0.9);
+            case SlayerUtils.VOIDGLOOM ->
+                    new Box(entity.getX() - 0.4, entity.getY() - 0.2, entity.getZ() - 0.4, entity.getX() + 0.4, entity.getY() - 3, entity.getZ() + 0.4);
+            case SlayerUtils.SVEN ->
+                    new Box(entity.getX() - 0.5, entity.getY() - 0.1, entity.getZ() - 0.5, entity.getX() + 0.5, entity.getY() - 1, entity.getZ() + 0.5);
+            default ->
+                    new Box(entity.getX() - 0.5, entity.getY() + 0.1, entity.getZ() - 0.5, entity.getX() + 0.5, entity.getY() - 2.2, entity.getZ() + 0.5);
+        };
+    }
+
+    /**
+     * <p> Finds the closest matching MobEntity for the armorStand using entityClass and armorStand age difference to filter
+     * out impossible candidates, returning the closest mob of those remaining in the search box by block distance </p>
+     *
+     * @param entityClass the mob type of the Slayer (i.e. ZombieEntity.class)
+     * @param armorStand  the entity that contains the display name of the Slayer (mini)boss
+     */
+    private static MobEntity findClosestMobEntity(Class<? extends MobEntity> entityClass, ArmorStandEntity armorStand) {
+        return armorStand.getWorld().getEntitiesByClass(entityClass, armorStand.getDimensions(null).getBoxAt(armorStand.getPos()).expand(1.5), entity -> !entity.isDead() && entity.age > armorStand.age - 4 && entity.age < armorStand.age + 4)
+                .stream()
+                .filter(entity -> !(entity instanceof CaveSpiderEntity)) // CaveSpider extends Spider so filter out mob for BroodFather highlight
+                .min(Comparator.comparingDouble((MobEntity e) -> e.distanceTo(armorStand)))
+                .orElse(null);
+    }
+
+    /**
+     * <p> Adds the Entity UUID to the Hashset of Slayer Mobs to glow </p>
+     *
+     * @param armorStand the entity that contains the display name of the Slayer (mini)boss
+     */
+    public static void setSlayerMobGlow(ArmorStandEntity armorStand) {
+        String slayerType = SlayerUtils.getSlayerType();
+        Class<? extends MobEntity> entityClass = SLAYER_MOB_TYPE.get(slayerType);
+        if (entityClass != null) {
+            MobEntity closestEntity = findClosestMobEntity(entityClass, armorStand);
+            if (closestEntity != null) MOBS_TO_GLOW.add(closestEntity.getUuid());
+        }
+    }
+
+    public static void onEntityDeath(Entity entity) {
+        MOBS_TO_GLOW.remove(entity.getUuid());
```

Comment:
Quite critical bug, please fix.

I believe #823 is caused by this. Please add a null check before calling `getUuid`. I have extracted the log below. Can you also add `@Nullable` to the `entity` parameter in `skyblocker$onEntityDeath`?

```java
[12:58:01] [Render thread/ERROR]: Failed to handle packet net.minecraft.class_2663@7cb18ae9
java.lang.NullPointerException: Cannot invoke "net.minecraft.class_1297.method_5667()" because "entity" is null
	at de.hysky.skyblocker.skyblock.slayers.SlayerMobs.onEntityDeath(SlayerMobs.java:127) ~[skyblocker-1.21.1-alpha-a61def4+1.21.jar:?]
	at net.minecraft.class_634.modifyExpressionValue$cge000$skyblocker$onEntityDeath(class_634.java:9104) ~[client-intermediary.jar:?]
	at net.minecraft.class_634.method_11148(class_634.java:1066) ~[client-intermediary.jar:?]
	at net.minecraft.class_2663.method_11471(class_2663.java:41) ~[client-intermediary.jar:?]
	at net.minecraft.class_2663.method_11054(class_2663.java:13) ~[client-intermediary.jar:?]
	at net.minecraft.class_2600.method_11072(class_2600.java:27) ~[client-intermediary.jar:?]
	at net.minecraft.class_1255.method_18859(class_1255.java:162) [client-intermediary.jar:?]
	at net.minecraft.class_4093.method_18859(class_4093.java:23) [client-intermediary.jar:?]
	at net.minecraft.class_1255.method_16075(class_1255.java:136) [client-intermediary.jar:?]
	at net.minecraft.class_1255.method_5383(class_1255.java:121) [client-intermediary.jar:?]
	at net.minecraft.class_310.method_1523(class_310.java:1240) [client-intermediary.jar:?]
	at net.minecraft.class_310.method_1514(class_310.java:882) [client-intermediary.jar:?]
	at net.minecraft.client.main.Main.main(Main.java:256) [client-intermediary.jar:?]
	at net.fabricmc.loader.impl.game.minecraft.MinecraftGameProvider.launch(MinecraftGameProvider.java:470) [fabric-loader-0.15.11.jar:?]
	at net.fabricmc.loader.impl.launch.knot.Knot.launch(Knot.java:74) [fabric-loader-0.15.11.jar:?]
	at net.fabricmc.loader.impl.launch.knot.KnotClient.main(KnotClient.java:23) [fabric-loader-0.15.11.jar:?]
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103) ~[?:?]
	at java.base/java.lang.reflect.Method.invoke(Method.java:580) ~[?:?]
	at oolloo.jlw.Wrapper.invokeMain(Wrapper.java:71) [JavaWrapper.jar:?]
	at oolloo.jlw.Wrapper.main(Wrapper.java:51) [JavaWrapper.jar:?]
```

Reply:
I've had 2 NullPointerException, the 1st was custom chat rule and the 2nd is this.
the 1st crash client, the 2nd frozen client, when is the 3rd? :cry: